### ISSUE TITLE
Chord Memory

### ISSUE THREAD

**Issue by matthieuroche** [surge-synthesizer/surge#1737]
**Background**
So, I am using my quarantine time to go through some tracks that I like, trying to rebuild them based on what I know from the gear possibly used by the artists. I have come to realize that many of the sounds that I am trying to reproduce at the moment were obtained with an alpha-juno, using its chord memory function.

**Feature request**
Would it be possible to add chord memory to Surge? 
Thanks for your attention!

Matthieu

**Comment by esaruoho** [comment]
Color me interested. I've also owned an Alphajuno1 since 2002 and loved it quite a bit.

**Comment by baconpaul** [comment]
Can you summarize what chord memory is for me? Or point at a good explainer?

**Comment by baconpaul** [comment]
Quick google makes me think it is a feature where if you select “1 5 7” and press c it plays c e g; and if you press d it pays d f# a. And you can choose any set of things. Kinda like on old organs. Is that right?

If that’s right, curious why you would put it in synth as opposed to writing a midi->midi vst3 or au which did the transformation (which might be a great addition to the surge plugin set).

Thanks!

**Comment by matthieuroche** [comment]
Hi Paul,

thanks for your quick reply, as always! Your google search is correct. I
understand your question, and I can see how this could easily be a task
performed by a VST/AU MIDI effect. There are already effects that do this
actually. However, I use Renoise as a host and it does not support MIDI
VST/AU effects. For example, I cannot use Kirnu Cream, a plug-in that has
this functionality, in a reliable fashion. That is why I was wondering
whether such a function could be added to Surge, as this synth becomes the
one I turn to to make most of my sounds.

My understanding is that this function was added (or at least used) to
expand the sonic possibilities of synths such as Alpha Junos or the Korg
Monopoly. Regarding the latter, you actually have to rely on chord memory
to obtain sounds that use multiple oscillators tuned to other intervals
than octaves (see for example the documentation of the MonoFury emulation
of the Monopoly by FBVC). This is just a tool that has been put to great
use by artists such as Drexciya for example, if you are familiar with their
music.

Cheers!

**Comment by baconpaul** [comment]
Right

Thinking of it: on unique “surge-y” way to do this is to have tuned unison. We recently added extended unison spreads (you can now actually make an octave uni spread if you RMB extend)

But another, of course, is to add midi transformations at the input. Adding the feature is super simple; adding the UI gestures and associated parameters is harder but the reason we are in the middle of a pretty big overhaul of the synth is to allow us to do things exactly like that. Will definitely keep this on the list for an upcoming release. Probably “not in the next month” though - we have a pretty deep book ahead of us and to do this properly requires us to finish some of those parameter and UI adjustments which are the basis of 1.7

**Comment by matthieuroche** [comment]
Hi Paul,

writing this morning my reply to you, I actually had exactly this idea that
tuned unison might do the trick, but I had not realized that Surge had an
extended unison tuning range. will give it a try!

Happy to read that it could be added to Surge anyway. Keep up the good
work, I am amazed at how good surge is!

Best

Matthieu

**Comment by baconpaul** [comment]
It’s a new feature we added in 1.6.6 (or maybe 1.6.5). Right mouse on the ‘unison range’ slider and choose ‘extend’ and it will go up to 1200 (cents) in relative mode and 192 (hz) in absolute mode (you can choose absolute also on the right mouse menu).

The problem is your can’t use it to make a major triad. But you could do unison = 3 and set the spread to 300 for a diminished triad or 1100 for some pretty cool quartal harmonies

**Comment by baconpaul** [comment]
(You can’t make a major triad since the voices uniformly span the space between 0 and the max)

**Comment by matthieuroche** [comment]
Actually tuned unison might be even funnier, with the ability to finely
tune the chord you are after. That might be very relevant when using osc
waveforms with unusual harmonic content (thinking about primes, etc...)

**Comment by baconpaul** [comment]
Yeah and don’t forget with full scl/kbm support you can do some amazing tuning things, since many of the tuning adjustments are in pitch (tuned) space not frequency space. (Although Unison is in untuned space I beleive). So if you have two oscillators and set th pitch to the second on ‘3’ it isn’t a minor third; it is whatever 3 Keys higher in your tuning is.

**Comment by matthieuroche** [comment]
Heheh, I did not explore this bit yet :) I am actually going through a
phase where I try to reproduce tracks that I like with a volutraily limited
set of soft synths and effects, and i did not try to recreate a tune in an
unusual key.

**Comment by mkruselj** [comment]
I don't think unison in extended mode is the right way to do this. You really want chord memory as its own thing. You play a chord to learn it, then can transpose it around with a single key. Benefit here is that learned chord can be a part of the preset.

**Comment by baconpaul** [comment]
Oh yeah it is clearly different than unison. (Also because you want a chord where arch note has independent unison). But unison exists today and chord memory doesn’t. 

It’s a neat idea. It got me thinking about other front-of-synth midi transforms we may want to do.

It is surprising how uneven support for pure midi plugins is in DAW space, also.

**Comment by matthieuroche** [comment]
Cool, I am happy that I am not the only one interested in this! If it can
lead to other MIDI goodness in Surge, this is even better! Thank you to you
both for making Surge an even better synth!

**Comment by mkruselj** [comment]
Might also be a good idea to have chord memory presets, like say 8 slots to which we could store chords, you could MIDI learn those buttons (I suppose they'd be sorta like CHSwitch2 style) or host automate the "chord memory slot" parameter to change memorized chords on the fly. Or maybe even modulate with step seq, say! Best thing - this stuff would all be a part of the patch.

**Comment by matthieuroche** [comment]
Fantastic idea!
